lib/src/function.dart

```diff
@@ -10,6 +10,11 @@ import 'token.dart';
 typedef PathFunction = String Function(Map<String, String> args);
 
 /// Creates a [PathFunction] from a [path] specification.
+///
+/// The returned function will always generate paths using the same case as the
+/// input [path]. If a path specification is being used to generate paths and
+/// to match case insensensitively, the canonical casing should be used in the
+/// specification.
```

Comment:
I think this goes without saying. I appreciate the extra documentation, but I don't see the need to associate this method with the caseSensitive parameter. Also, I'm not sure this advice is necessary relevant for everyone, it seems like a decision the end user can choose to make.